src/coreclr/jit/lowerxarch.cpp

```diff
@@ -9190,134 +9183,118 @@ bool Lowering::IsContainableHWIntrinsicOp(GenTreeHWIntrinsic* parentNode, GenTre
-
-                        assert((sizeof_simdType == 16) || (sizeof_simdType == 32));
-                        assert((sizeof_baseType == 1) || (sizeof_baseType == 2) || (sizeof_baseType == 4));
-
-                        const unsigned expectedSize = sizeof_simdType / (sizeof_baseType * 2);
-                        const unsigned operandSize  = genTypeSize(childNode->TypeGet());
-
-                        assert((sizeof_simdType != 16) || (expectedSize == 8) || (expectedSize == 4) ||
-                               (expectedSize == 2));
-                        assert((sizeof_simdType != 32) || (expectedSize == 16) || (expectedSize == 8) ||
-                               (expectedSize == 4));
-
-                        supportsGeneralLoads = (operandSize >= expectedSize);
-                    }
-                    else
-                    {
-                        // The memory form of this already takes a pointer and should be treated like a MemoryLoad
-                        supportsGeneralLoads = !childNode->OperIsHWIntrinsic();
+                        default:
+                        {
+                            break;
+                        }
                     }
-                    break;
-                }
-
-                case NI_SSE2_ConvertToVector128Double:
-                case NI_AVX_ConvertToVector256Double:
-                case NI_AVX512F_ConvertToVector512Double:
-                case NI_AVX512F_VL_ConvertToVector128Double:
-                case NI_AVX512F_VL_ConvertToVector256Double:
-                case NI_AVX10v1_ConvertToVector128Double:
-                case NI_AVX10v1_ConvertToVector256Double:
-                {
-                    assert(!supportsSIMDScalarLoads);
-
-                    // Most instructions under the non-VEX encoding require aligned operands.
-                    // Those used for Sse2.ConvertToVector128Double (CVTDQ2PD and CVTPS2PD)
-                    // are exceptions and don't fail for unaligned inputs as they read half
-                    // the vector width instead
-
-                    supportsAlignedSIMDLoads   = !comp->opts.MinOpts();
-                    supportsUnalignedSIMDLoads = true;
 
-                    const unsigned expectedSize = genTypeSize(parentNode->TypeGet()) / 2;
                     const unsigned operandSize  = genTypeSize(childNode->TypeGet());
+                    const unsigned expectedSize = genTypeSize(parentNode->TypeGet()) / widenFactor;
 
-                    if (childNode->OperIsConst() || childNode->isMemoryOp())
+                    if (expectedSize < 16)
                     {
-                        // For broadcasts we can only optimize constants and memory operands
-                        // since we're going from a smaller base type to a larger base type
-                        supportsGeneralLoads = supportsUnalignedSIMDLoads && (operandSize >= expectedSize);
-                    }
```

Comment:
Deleting this restriction led to some interesting diffs in MinOpts.  ex:

```diff
@@ -31,15 +31,14 @@ G_M4722_IG02:        ; bbWeight=1, gcrefRegs=0000 {}, byrefRegs=0000 {}, byref
        ; gcrRegs +[rsi]
        call     [<unknown method>]
        ; gcrRegs -[rsi]
-       vmovaps  xmm0, xmmword ptr [rbp+0x10]
-       vcvtps2pd ymm0, ymm0
+       vcvtps2pd ymm0, xmmword ptr [rbp+0x10]
        vmovups  ymmword ptr [rbp-0x30], ymm0
        mov      rax, bword ptr [rbp-0x08]
        ; byrRegs +[rax]
        vmovups  ymm0, ymmword ptr [rbp-0x30]
        vmovups  ymmword ptr [rax], ymm0
        mov      rax, bword ptr [rbp-0x08]
-						;; size=52 bbWeight=1 PerfScore 21.50
+						;; size=48 bbWeight=1 PerfScore 20.50
 G_M4722_IG03:        ; bbWeight=1, epilog, nogc, extend
        vzeroupper 
        add      rsp, 176
```

Here, the managed code is 

https://github.com/dotnet/runtime/blob/1e4140457df4a5d3c647ecf263b0057d84dbc210/src/libraries/System.Private.CoreLib/src/System/Runtime/Intrinsics/VectorMath.cs#L2736

Which doesn't have an explicit aligned load, however the current logic prevents containment for this normal indir.  JIT just happens to emit an aligned load because it knows the value is aligned on the stack.  We're now able to contain it.